### PR TITLE
Spinback fix

### PR DESCRIPTION
**Pull request by ronso0**
Fixes a few spinback "regressions" noticed in #4705
* fix infinite acceleration (never comes to halt currently)
* spin**back** when interrupting softStart (was spinning forward)
* allow spinback on stopped deck

The first two are fixed by a single `-` 7fed042f29a1ed8b1f2ea62ccd1bc3f2c9bf2cf1 (after hours trying to understand the alpha-beta scratch filter).

Allowing spinback on stopped deck (IIRC that was working a few years back) was more work as it required cleaning up `brake()`, and I hope someone has motivation to review it so we can merge it to "stable" 2.3

- [x] remove debug output / use kLogger?

### CODE REVIEW

**Comment by ronso0** [comment]
still draft because I'd like to keep the debug output 87c23c210004d531f8c425bd27de59248f2e8af6 until review is finished.

**Comment by ronso0** [comment]
Also: merging this to `main` requires manual editing since there are conflicts with renamed and moved files ( + clang-format, no actual behaviour or syntax changes).
Please ping me before merging so I can make sure I have time to merge to main !!

**Review by daschuer — COMMENTED** [review]
Some more comments. I hope I will find time for a manual test soon.

**Comment by ronso0** [comment]
> Some more comments. I hope I will find time for a manual test soon.

I am testing with the GUI and a MIDI Through mapping that uses callbacks of toggles on deck 2:
https://gist.github.com/ronso0/70bc239c14c1de491dbb87dc303a789f
Eject = init brake
Repeat = init spinback
Slip mode = init softstart

**Comment by ronso0** [comment]
Thanks, I pushed th small fixes.

**Comment by daschuer** [comment]
I have tested it and I think it works like desired. 
So I think it is ready to polish for merge. 

I must admit hat I have never used the backspin effect before and I think it is implemented unnaturally. 
My expectation was that it sound like pushing a vinyl backward with a sharp acceleration an a spin down phase afterwards. Something like demonstrated here: https://www.youtube.com/watch?v=dzEJoSkwvnI at 0:20 
Is this a topic of this PR?

**Comment by ronso0** [comment]
Exactly, that's how a backspin is supposed to sound like.
I am sure it was initially implemented like this (I simply doubt someone would implement it in such a weird way as it is now), so I suspect something broke since then, but I didn't figure out _when_ with a quick git search.

Actually I think some of the debug output is quite helpful, so I'll remove some of it and put the helpful part behind #ifdef SCRATCH_DEBUG 
Okay?

**Comment by ronso0** [comment]
Thanks for testing btw, appreciated!

**Comment by daschuer** [comment]
Yes OK, thank you.

**Comment by daschuer** [comment]
I will file a bug for the implementation issue. Maybe we find a GSoC contributor taking care with @ywwg.

**Comment by daschuer** [comment]
https://bugs.launchpad.net/mixxx/+bug/1967715

**Comment by ronso0** [comment]
Last commit disables the debug output in real-time code.
Ready for merge.

File: src/controllers/controllerengine.cpp [hunk 1]
```diff
@@ -1333,31 +1372,47 @@ void ControllerEngine::scratchTick(int deck, int interval) {
...
 
+#if SCRATCH_DEBUG_OUTPUT
     const double oldRate = filter->predictedVelocity();
+#endif
 
     // Give the filter a data point:
 
     // If we're ramping to end scratching and the wheel hasn't been turned very
     // recently (spinback after lift-off,) feed fixed data
+    qDebug() << ".";
```
Comment (daschuer) [inline src/controllers/controllerengine.cpp:1397]: Her is a leftover

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/controllers/controllerengine.cpp (hunk 1 above)
Comment (ronso0) [reply src/controllers/controllerengine.cpp:1397]: whoops, fixed.

**Review by daschuer — APPROVED** [review]
Thank you. LGTM

**Pull request merged**